src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,151 @@
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * The Parser class takes in inputs from the user and convert into commands.
+ */
+public class Parser {
+
+    private static final List<String> DATE_FORMATS = Arrays.asList("yyyy/MM/dd HHmm", "y/M/d HHmm", "y-M-d HHmm");
+    private static final String BYE = "bye";
+    private static final String LIST = "list";
+    private static final String DONE = "done";
+    private static final String TODO = "todo";
+    private static final String EVENT = "event";
+    private static final String DEADLINE = "deadline";
+    private static final String DELETE = "delete";
+    private static final String FIND = "find";
+
+    /** Returns the command based on user input. */
+    public static Command parse(String input) throws DukeException {
+        int i = input.trim().indexOf(' ');
+        String command = input;
+        String detail = "";
+        if (i > 0) {
+            command = input.substring(0, i);
+            detail = input.substring(i).trim();
+        }
+
+        switch (command) {
+        case BYE:
+            return new ByeCommand();
+        case LIST:
+            return new ListCommand();
+        case DONE:
+            return new DoneCommand(parseNumber(detail));
+        case TODO:
+            return new TodoCommand(parseTodo(detail));
+        case EVENT:
+            return new EventCommand(parseEvent(detail));
+        case DEADLINE:
+            return new DeadlineCommand(parseDeadline(detail));
+        case DELETE:
+            return new DeleteCommand(parseNumber(detail));
+        case FIND:
+            return new FindCommand(detail.trim());
+        default:
+            throw new DukeException("Oops! I'm sorry, but I don't know what that means");
+        }
+    }
+
+    /**
+     * Parse the date given by user.
+     * @param dateString Date provided by user in the form of string.
+     * @return Date in LocalDateTime.
+     */
+    public static LocalDateTime parseDate(String dateString) {
+        for (String format : DATE_FORMATS) {
+            try {
+                return LocalDateTime.parse(dateString, DateTimeFormatter.ofPattern(format));
+            } catch (DateTimeParseException e) {
+                e.getMessage();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return the ToDo the user specified.
+     * @param input The todo details given by user.
+     * @return A ToDo with the input given.
+     */
+    public static ToDo parseTodo(String input) throws DukeException {
+        if (input.equals("")) {
+            throw new DukeException("Oops! Todo cannot be empty");
```

Comment:
I'm a bit confused here... maybe can change the variable "input" to "detail" like in the parse method because for some reason i assumed the input is the whole string ><